### DISCUSSION TITLE
parse paragraph with line breaks

### DISCUSSION THREAD

**Discussion by denisso** [remarkjs/remark#1132] [Q&A]
my code
```js
import { unified } from "unified";
import remarkParse from "remark-parse";
import rehypeStringify from "rehype-stringify";
import remarkRehype from "remark-rehype";

function remarkParagragh() {
  return (tree) => {
/*
tree
  {
    type: "text",
    value: "1\r\n2\r\n3",
    position: {
      start: {
        line: 1,
        column: 1,
        offset: 0,
      },
      end: {
        line: 3,
        column: 2,
        offset: 7,
      },
    },
  },
]
*/
  };
}
  result = unified()
    .use(remarkParse) 
    .use(remarkParagragh)
    .use(remarkRehype)
    .use(rehypeStringify)
    .processSync(buffer);
```

markdown buffer
```
1
2
3
```

html output 
```html
<p>1
2
3</p>
```

i kwon about plugin remark-breaks, but **i need three paragraphs**

```html
<p>1</p>
<p>2</p>
<p>3</p>
```

to make additional lines in markdown, the option does not suit me either
```
1

2

3
```

I can write a plugin that will split a paragraph into three paragraphs, which is connected after the markdown parser. But it doesn't seem optimal to me. Because it feels like a double job. 

Can I somehow add my paragraph parser at the parsing stage? 
I didn't find any options in remark parse to connect my handlers, or an option to work with streak breaks.

**Comment by remcohaszing** [comment]
I highly recommend to not invent your own syntax. It's best to represent paragraphs using actual markdown paragraphs.

**Reply by denisso** [reply]
its need for only custom work

**Reply by wooorm** [reply]
It probably really isn’t. It probably doesn’t relate to markdown.

If it does, you need to explain it.

**Reply by denisso** [reply]
Thank you for responding so quickly

**Comment by denisso** [comment]
Excuse me, I hurried to create a topic.

```html
<p>
	<span class="uniqId">1</span>
	<br>
	<span class="uniqId">2</span>
	<br>
	<span class="uniqId">3</span>
</p>
```

works well for me

**Reply by remcohaszing** [reply]
Beware that line breaks are typically semantically incorrect in writing. The only legitimate use case that comes to mind is song lyrics / poetry.

I understand it's convenient for layout in markdown though, I've used it myself too sometimes. However, it's better to use a `<div>` for layout if you can.
